src/iframeMessenger.js

```diff
@@ -423,6 +423,49 @@
             }
         }
 
+        function _addAcquisitionDataToLinks(acquisitionData) {
+            var links = document.getElementsByClassName('js-acquisition-link');
+            var json = encodeURIComponent(JSON.stringify(acquisitionData));
+            for (var i = 0; i < links.length; i++) {
+                var href = link.getAttribute('href');
+                if (href) {
+                    href += href.indexOf('?') === -1 ? '?' : '&';
+                    href += 'acquisitionData=';
+                    href += json;
+                    link.setAttribute('href', href);                    
+                }
+            }
+        }
+
+        function _buildAcquisitionData(acquisitionData, referrerData) {
+            if (acquisitionData && referrerData) {
+                var out = {};
+                out.componentId = acquisitionData.componentId;
+                out.componentType = acquisitionData.componentType;
+                out.source = acquisitionData.source;
+                out.abTest = acquisitionData.abTest;
+                out.campaignCode = acquisitionData.campaignCode;
+                out.referrerPageviewId = referrerData.referrerPageviewId;
+                out.referrerUrl = referrerData.referrerUrl;
+                return out;
+            }
+            return null;
+        }
+
+        function _enrichAcquisitionLinks(acquisitionData) {
```

Comment:
might be worth documenting the expected shape of the object in a comment above the function too, as you've very helpfully done in the readme - in case someone dives straight into the code without scrolling through the readme